### ISSUE TITLE
loss curve?

### ISSUE THREAD

**Issue by AndyYuan96** [facebookresearch/ContrastiveSceneContexts#22]
Hi, I just wonder whether you can share the training loss curve of the project, as I now can run the code, thank you very much.

**Comment by Sekunde** [comment]
Hi, 

please check out [this](https://tensorboard.dev/experiment/2O3rr40vRQa3XBDZrbNwAw/#scalars&_smoothingWeight=0.99) for the pre-train loss curves as well as the [config(http://kaldir.vc.in.tum.de/3dsis/contrastive_scene_contexts/pretrain/partition8_4096_100k.log). 

For the curves of downstream tasks, please refer to the respect links in the README.

**Comment by AndyYuan96** [comment]
> Hi,
> 
> please check out [this](https://tensorboard.dev/experiment/2O3rr40vRQa3XBDZrbNwAw/#scalars&_smoothingWeight=0.99) for the pre-train loss curves as well as the [config](http://kaldir.vc.in.tum.de/3dsis/contrastive_scene_contexts/pretrain/partition8_4096_100k.log).
> 
> For the curves of downstream tasks, please refer to the respective links in the README.

thank you very much，sekunde，do you have pointcontrast tensorboard log，I'm trying to apply pointcontrast method to lidar data, and change point to voxel format for lidar detector,  I find the loss curve is a little wired, the loss drop very slowly, as I don't have scannet data, so I don't train the paper code, after looking your tensor board, the loss did drop very slowly.

**Comment by AndyYuan96** [comment]
> Hi,
> 
> please check out [this](https://tensorboard.dev/experiment/2O3rr40vRQa3XBDZrbNwAw/#scalars&_smoothingWeight=0.99) for the pre-train loss curves as well as the [config](http://kaldir.vc.in.tum.de/3dsis/contrastive_scene_contexts/pretrain/partition8_4096_100k.log).
> 
> For the curves of downstream tasks, please refer to the respective links in the README.

Hi, sekunde, I just wonder whether you meet the situation that the positive pair's distance increase very quick to around 1 in the begin of training，I apply pointcontrast based method to lidar data，I find the only difference is my positive pair's distance increase too quick compared to pointcontrast

**Comment by Sekunde** [comment]
Hi, there

I did train pointcontrast long time ago, check [this](https://tensorboard.dev/experiment/iwnMgnaOTPqixi9nyxovtw/#scalars&_smoothingWeight=0.99) out.

what do you exactly mean by the "the positive pair's distance increase very quick to around 1"? I normally only see the loss.

**Comment by AndyYuan96** [comment]
> Hi, there
> 
> I did train pointcontrast long time ago, check [this](https://tensorboard.dev/experiment/iwnMgnaOTPqixi9nyxovtw/#scalars&_smoothingWeight=0.99) out.
> 
> what do you exactly mean by the "the positive pair's distance increase very quick to around 1"? I normally only see the loss.

assuming we choose 1024 point to train,  the point feature's normalized, so we want positive pair's two point feature's dot product to be closed to 1, and negative pair's two point feature's dot product to be closed to -1.So I statistics e^(positive pair distance) and Sum(e^(negative pair distance)) when training,  and I find my loss did decrease very quickly compared with pointcontrast loss, and I find that the reason is my e^(positive pair distance) very quickly go around e^(1) when training, just  around 100 iteration, so my loss decrease very quickly in the beginning of training compared with pointcontrast, and I don't know why my positive pair's distance go around 1 so quickly in the beginning.

**Comment by Sekunde** [comment]
Hi,  

Unfortunately, I never met that situation, normally the loss drops slowly in indoor data and continues decreasing a bit. As the nature of indoor data and outdoor data is quite different (outdoor data is sparse and in larger range), you might need to adjust some hyper-parameters, e.g. the temperature.

It seems to me that your pre-train task is too easy to optimize (loss drops too fast in the first iterations), e.g. too easy to distinguish the positive and negative pairs? It is not recommended to make pre-train task too easy as it might not learn too much meaningful features.

Maybe you sampled too many points on the road surface? You can try to sample the points on the cars/pedestrians or sampling the pairs of points spatially closer to each other? I am not exactly sure if that is the reason...but hope it helps to make your loss not drop too fast.

**Comment by Hiusam** [comment]
> Hi,
> 
> please check out [this](https://tensorboard.dev/experiment/2O3rr40vRQa3XBDZrbNwAw/#scalars&_smoothingWeight=0.99) for the pre-train loss curves as well as the [config](http://kaldir.vc.in.tum.de/3dsis/contrastive_scene_contexts/pretrain/partition8_4096_100k.log).
> 
> For the curves of downstream tasks, please refer to the respective links in the README.

Hi!
I like your work very much. I have several questions.
1. I want to ask why you set misc.train_num_thread=2? In your code, the num_workers should be `train_num_thread / num_gpus`, so if you set train_num_thread=2 and num_gpus=8, the num_works will be 0 and the data time will be very high, which is what actually happened when I experimented with this setting (data time is 7s). 
2. I have to set train_num_thread=16, in which case, the data time is about 0.03 and the train time is about 4.8. The data time is normal but the train time seems to be 2s higher than that in your log. I wonder why. I am also using 8 V100 GPUS on one node.

**Comment by Sekunde** [comment]
@Hiusam Hi, thanks for interests on our work.

1. the code has been refactored before releasing, so please change the train_num_thread so that it lower your data loading speed most.

2. We also notice the training time change for some time, I checked some logs (see below) it varies from 2.x to 4.x secs from node to node (we use slurm, so every time we are allocated with a different node); but in most times it should be around 2-3 secs, have you tried another node? One tip: you can turn data.world_space=False to see if it helps.

<img width="320" alt="Screenshot 2022-01-05 at 12 43 04" src="https://user-images.githubusercontent.com/6199161/148354232-0b8dd335-05c2-4160-ad43-3214bfed0c38.png">
<img width="320" alt="Screenshot 2022-01-05 at 12 46 43" src="https://user-images.githubusercontent.com/6199161/148354241-97724475-b813-4b9f-98a9-6f2dccfedc71.png">

**Comment by Hiusam** [comment]
Turning data.world_space=False doesn't help. T T

**Comment by Rezamh13** [comment]
> 

Hi @AndyYuan96, I have the same situation where the loss decreased very quickly and then hits a plateau. Have you find out how to solve it?

**Comment by zhang-zihui** [comment]
Hi, @Sekunde , I use the default hyper-parameters of pretraining and run the pretrain_sparseconv.sh file. But my loss is much larger than the loss curve provided by you.
![image](https://user-images.githubusercontent.com/120791554/233077087-874c5e4f-8256-4ff5-be08-1d5aa0452e23.png)

**Comment by zhang-zihui** [comment]
and I didn't find "mintor loss" in the code
